include/net/net_tls.h

```diff
@@ -0,0 +1,105 @@
+/** @file
+ * @brief Network TLS definitions
+ *
+ * An API for applications to configure TLS credentials.
+ */
+
+/*
+ * Copyright (c) 2018 Intel Corporation
+ * Copyright (c) 2018 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef __NET_TLS_H
+#define __NET_TLS_H
+
+/**
+ * @brief Network TLS management
+ * @defgroup net_tls Network TLS management
+ * @ingroup networking
+ * @{
+ */
+
+#include <stddef.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/** TLS credential type */
+enum net_tls_credential_type {
+	NET_TLS_CREDENTIAL_UNUSED,
+	NET_TLS_CREDENTIAL_CA_CERTIFICATE,
+	NET_TLS_CREDENTIAL_PSK,
+	NET_TLS_CREDENTIAL_PSK_ID
+};
+
+/** Secure tag, a reference for TLS credential */
+typedef int sec_tag_t;
+
+#if defined(CONFIG_NET_TLS) || defined(CONFIG_NET_DTLS)
+
+/**
+ * @brief Add a TLS credential.
+ *
+ * @details This function adds a TLS credential, that can be used
+ *          by TLS/DTLS for authentication.
+ *
+ * @param tag     A security tag that credential will be referenced with.
+ * @param type    A TLS/DTLS credential type.
+ * @param cred    A TLS/DTLS credential.
+ * @param credlen A TLS/DTLS credential length.
+ *
+ * @return 0 if ok, < 0 if error.
+ */
+int net_tls_credential_add(sec_tag_t tag, enum net_tls_credential_type type,
+			   const void *cred, size_t credlen);
+
+/**
+ * @brief Get a TLS credential.
+ *
+ * @details This function gets an already registered TLS credential,
+ *          referenced by @p tag secure tag of @p type.
+ *
+ * @param tag     A security tag of requested credential.
+ * @param type    A TLS/DTLS credential type of requested credential.
+ * @param cred    A TLS/DTLS credential.
+ * @param credlen A TLS/DTLS credential length.
+ *
+ * @return 0 if ok, < 0 if error.
+ */
+int net_tls_credential_get(sec_tag_t tag, enum net_tls_credential_type type,
```

Comment:
Generally, it's clear that access operations go int set/get pairs, but here we're talking about *security credentials*. What would be a usecase for a user application to get a system-configured private key for example? I'd double-think whether "get" operation should be part of public APIm and if so, would still mention that access restrictions may apply (and document error code returned in this case). Well, I guess that's more up to security folks like @d3zd3z .